Libs/RaptureCore/src/main/java/rapture/kernel/DocApiImpl.java

```diff
@@ -210,7 +218,15 @@ public void deleteDocRepo(CallingContext context, String docRepoUri) {
         }
         // Yeah this is like "delete everything that is prefixed by //docRepoUri
         SearchPublisher.publishDropMessage(context, internalUri.toString());
-        DocumentRepoConfigStorage.deleteByAddress(internalUri, context.getUser(), "Drop document repo");
+        // We can't just delete the repo. If we do then
+        // DocRepoCache.reloadConfig will just create it.
+        // So mark the config as having been deleted.
+        // DocumentRepoConfigStorage.deleteByAddress(internalUri,
+        // context.getUser(), "Drop document repo");
+        DocumentRepoConfig drc = DocumentRepoConfigStorage.readByAddress(internalUri);
+        drc.setConfig(DELETED);
```

Comment:
this seems a bit odd.  You're using the config field not for its intended purpose.  It would be cleaner to have a boolean field 'deleted' added to the DocumentRepoConfig object in types.api.  You can then check isDeleted, instead of doing string comparisons